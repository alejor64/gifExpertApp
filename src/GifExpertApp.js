import React, { Fragment, useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <Fragment>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr/>

            <ol>
                {
                    categories.map(category => {
                        return <GifGrid key={category} category={category} />
                    })
                }
            </ol>
        </Fragment>
    )
}

export default GifExpertApp