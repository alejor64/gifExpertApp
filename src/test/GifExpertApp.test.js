import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Pruebas en el componente <GifExpertApp />', () => {
    test('Debe renderizar el compoenente', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Dragon Ball', 'Naruto']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
    
})
