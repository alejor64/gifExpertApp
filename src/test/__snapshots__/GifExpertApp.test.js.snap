// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe renderizar el compoenente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
